ktor-client/ktor-client-plugins/ktor-client-resources/common/src/io/ktor/client/plugins/resources/builders.kt

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.plugins.resources
+
+import io.ktor.client.*
+import io.ktor.client.plugins.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.resources.*
+import io.ktor.client.request.delete as deleteBuilder
+import io.ktor.client.request.get as getBuilder
+import io.ktor.client.request.head as headBuilder
+import io.ktor.client.request.options as optionsBuilder
+import io.ktor.client.request.post as postBuilder
+import io.ktor.client.request.prepareDelete as prepareDeleteBuilder
+import io.ktor.client.request.prepareGet as prepareGetBuilder
+import io.ktor.client.request.prepareHead as prepareHeadBuilder
+import io.ktor.client.request.prepareOptions as prepareOptionsBuilder
+import io.ktor.client.request.preparePost as preparePostBuilder
+import io.ktor.client.request.preparePut as preparePutBuilder
+import io.ktor.client.request.prepareRequest as prepareRequestBuilder
+import io.ktor.client.request.put as putBuilder
+import io.ktor.client.request.request as requestBuilder
+
+/**
+ * Executes a [HttpClient] GET request, with url built from [resource]
+ */
+public suspend inline fun <reified T : Any> HttpClient.get(
```

Comment:
Should the upper bound be more specific here? Will we have a conflict with `get(URL)`?

Reply:
No, `Url` and `String` are more specific types and there will be no conflict. I have a test that checks exactly this case.